pkg/controller/endpointslice/reconciler.go

```diff
@@ -262,14 +266,23 @@ func (r *reconciler) reconcileByAddressType(service *corev1.Service, pods []*cor
 	}
 
 	if r.topologyCache != nil && hintsEnabled(service.Annotations) {
-		slicesToCreate, slicesToUpdate = r.topologyCache.AddHints(si)
+		slicesToCreate, slicesToUpdate, events = r.topologyCache.AddHints(si)
 	} else {
 		if r.topologyCache != nil {
+			if r.topologyCache.HasPopulatedHints(si.ServiceKey) {
+				events = append(events, &topologycache.EventBuilder{
+					EventType: corev1.EventTypeWarning,
+					Reason:    "TopologyAwareHintsDisabled",
+					Message:   fmt.Sprintf("Topology Aware Hints has been disabled, removing %s hints from service %q", addressType, si.ServiceKey),
```

Comment:
because we do not really put hints on a service, I want to change wording "removing hints from service" to "removing hints for service" 
wdyt @robscott ?

Reply:
SGTM